src/nspawn/nspawn.c

```diff
@@ -5710,7 +5710,7 @@ static int run(int argc, char *argv[]) {
                 if (r < 0)
                         goto finish;
 
-                if (!arg_verity_settings.root_hash && dissected_image->has_verity)
+                if (!arg_verity_settings.root_hash && dissected_image->has_verity && !dissected_image->has_verity_sig)
                         log_notice("Note: image %s contains verity information, but no root hash specified! Proceeding without integrity checking.", arg_image);
```

Comment:
Log message is misleading here: integrity checking will be used but without cryptographic validation

Reply:
actually we won't do integrity checking at all in this case here. But the log message could use some love indeed.

i.e. when we have neither roothash nor embedded signature data (from which we can parse the root hash) we really have nothing to go by, we cannot set dm-verity in that case, hence we don't.